test/behavior/bitcast.zig

```diff
@@ -11,21 +12,119 @@ test "@bitCast i32 -> u32" {
 }
 
 fn testBitCast_i32_u32() !void {
-    try expect(conv(-1) == maxInt(u32));
-    try expect(conv2(maxInt(u32)) == -1);
+    try expect(conv_i32(-1) == maxInt(u32));
+    try expect(conv_u32(maxInt(u32)) == -1);
+    try expect(conv_u32(0x8000_0000) == minInt(i32));
+    try expect(conv_i32(minInt(i32)) == 0x8000_0000);
```

Comment:
no maxInt(i32), minInt(u32), -0 etc?
Is it possible to test error cases?

Reply:
1. out of range stuff for longer aligned integer could be broken.
one option to test this is to write a bit pattern to a buffer, cast a subpart (or whole) region to an exotic integer and test, if the casted integer fails the bitcast.

2. another option is to test, if the bitcast only applies to a subpart of a buffer by writing to it and check expected results with the manual combination of bit masks or their results.

Option 2 with bit mask (fun) feels like it is in scope of this PR, Option 1 feels like it belong into the other PR.

Reply:
> write undefined values to the output buffer/integer, if we're concerned about incomplete updates

Uh, I think I got confused with `@truncate` and was thinking `@bitCast` could do the same.
Is there an operation to get a subslice at bit-boundary, which `@bitCast` can operate on or are all integers
aligned at byte boundary (via `@truncate`)?

(probably not relevant)
Then it would look like 1. Use `const buf: []const u8 : .{ bitpattern};`, 2. take a subslice, 3. convert to the exotic integer, 4. make an arithmetic operation (negation, abs, shift left/right, addition, subraction, multiplication, division) and 5. compare to the expected bitpattern of the whole slice `buf`.

Mark this as resolved, if you think this does not apply.

Reply:
Ah, no problem.

I don't think this applies - read/writeTwosComplement have no functionality to read/write a subset of the bits backing a big integer